features/cellular/framework/API/CellularDevice.h

```diff
@@ -222,14 +222,22 @@ class CellularDevice {
     /** Stop the current operation. Operations: set_device_ready, set_sim_ready, register_to_network, attach_to_network
      *
      */
+    MBED_DEPRECATED_SINCE("mbed-os-5.15", "Use CellularDevice::shutdown() instead.")
     void stop();
 
     /** Get the current FileHandle item used when communicating with the modem.
      *
      *  @return reference to FileHandle
      */
+    MBED_DEPRECATED_SINCE("mbed-os-5.15", "Use CellularDevice::get_filehandle() instead.")
     FileHandle &get_file_handle() const;
 
+    /** Get the current filehandle.
+     *
+     *  @return pointer to filehandle or NULL
```

Comment:
When will this return null? As I recall part of the reason for original API was that it wouldn't.

Reply:
Filehandle is used to identify serial link and/or data channels. When CellularDevice is over a non-filehandle alike link, it's possible that filehandle is NULL since data channels may not be immediately available.

Reply:
Sounds like some sort of redesign being planned? Got any more info?

Deprecations usually accompany or follow redesigns, not precede them - it's a bit hard to review a deprecation that's a hint of "things to come".

At the minute, the FileHandle is part of the fundamental channel ownership. Are you planning to increase complexity here by letting lower levels kill themselves/not exist while the higher levels remain active?

At the minute, we're generally preferring the "suspend/resume" API for that sort of thing, as that allows you to retain ownership.

Reply:
Marked `CellularDevice::get_file_handle` as deprecated.

Reply:
@kjbracey-arm the update resolved this?